packages/core-data/src/resolvers.js

```diff
@@ -273,34 +273,49 @@ export const getEmbedPreview =
  * Checks whether the current user can perform the given action on the given
  * REST resource.
  *
- * @param {string}  action   Action to check. One of: 'create', 'read', 'update',
- *                           'delete'.
- * @param {string}  resource REST resource to check, e.g. 'media' or 'posts'.
- * @param {?string} id       ID of the rest resource to check.
+ * @param {string}  requestedAction Action to check. One of: 'create', 'read', 'update',
+ *                                  'delete'.
+ * @param {string}  resource        REST resource to check, e.g. 'media' or 'posts'.
+ * @param {?string} id              ID of the rest resource to check.
  */
 export const canUser =
-	( action, resource, id ) =>
-	async ( { dispatch } ) => {
-		const methods = {
-			create: 'POST',
-			read: 'GET',
-			update: 'PUT',
-			delete: 'DELETE',
-		};
-
-		const method = methods[ action ];
-		if ( ! method ) {
-			throw new Error( `'${ action }' is not a valid action.` );
+	( requestedAction, resource, id ) =>
+	async ( { dispatch, registry } ) => {
+		const { hasStartedResolution } = registry.select( STORE_NAME );
+
+		let resourcePath;
+		let retrievedActions;
+		if ( id ) {
+			retrievedActions = [ 'create', 'read', 'update', 'delete' ];
+			resourcePath = `${ resource }/${ id }`;
+
+			// Prevent resolving the same resource twice.
+			for ( const relatedAction of retrievedActions ) {
+				if ( relatedAction === requestedAction ) {
+					continue;
+				}
+				const isAlreadyResolving = hasStartedResolution( 'canUser', [
+					relatedAction,
+					resource,
+					id,
+				] );
+				if ( isAlreadyResolving ) {
+					return;
+				}
+			}
+		} else {
+			retrievedActions = [ 'create' ];
```

Comment:
```suggestion
			retrievedActions = [ 'create', 'read' ];
```

The `read` action can also be used without an ID - `canUser( 'read', 'settings' )`.

Reply:
Good spot!